pkg/operator/latencyprofilecontroller/status_utils.go

```diff
@@ -0,0 +1,74 @@
+package latencyprofilecontroller
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	reasonWorkerLatencyProfileUpdated = "ProfileUpdated"
+	reasonWorkerLatencyProfileEmpty   = "ProfileEmpty"
+)
+
+// setWLPStatusCondition is used to set condition in config node object status.workerLatencyProfileStatus
+func setWLPStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Condition) {
+	if conditions == nil {
+		conditions = &[]metav1.Condition{}
+	}
+	existingCondition := findWLPStatusCondition(*conditions, newCondition.Type)
+	if existingCondition == nil {
+		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		*conditions = append(*conditions, newCondition)
+		return
+	}
+
+	if existingCondition.Status != newCondition.Status {
+		existingCondition.Status = newCondition.Status
+		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
+	}
+
+	existingCondition.Reason = newCondition.Reason
+	existingCondition.Message = newCondition.Message
+}
+
+// FindWLPStatusCondition is used to find conditions from config node object status.workerLatencyProfileStatus
+func findWLPStatusCondition(conditions []metav1.Condition, conditionType string) *metav1.Condition {
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return &conditions[i]
+		}
+	}
+
+	return nil
+}
+
+func (c *LatencyProfileController) updateConfigNodeStatus(newCondition metav1.Condition) (updated bool, err error) {
```

Comment:
I think you could use `operatorv1helpers.UpdateStatus`

Reply:
Initially, was planning to use the same: operatorv1helpers.UpdateStatus but seems like we cannot, because that expects an OperatorClient and it's status object need to be operatorv1.Condition whereas in our case we're configv1.Node uses metav1.Condition in its status conditions. (hard luck till we have strong support for go generics I guess 😀)

Essentially, I copied over majority of the logic from there itself to attain similar status updation behaviour.